aidial_adapter_bedrock/llm/converse/input.py

```diff
@@ -0,0 +1,325 @@
+                storage
+            )
+            content.append(
+                {
+                    "image": {
+                        "format": to_converse_image_type(resource.type),
+                        "source": {"bytes": resource.data},
+                    }
+                }
+            )
+    if message.function_call and message.tool_calls:
+        raise ValidationError(
+            "You cannot use both function call and tool calls in the same message"
+        )
+    elif message.function_call:
+        content.append(function_call_to_content_part(message.function_call))
+    elif message.tool_calls:
+        content.extend(
+            [
+                tool_call_to_content_part(tool_call)
+                for tool_call in message.tool_calls
+            ]
+        )
+
+    return content
+
+
+async def to_converse_message(
+    message: DialMessage,
+    storage: FileStorage | None,
+    supported_image_types: list[str] | None = None,
+) -> ConverseMessage:
+
+    return {
+        "role": to_converse_role(message.role),
+        "content": await _get_converse_message_content(
+            message, storage, supported_image_types
+        ),
+    }
+
+
+@dataclass
+class ExtractSystemPromptResult:
+    system_prompt: ConverseTextPart | None
+    system_message_count: int
+    modified_messages: List[DialMessage]
+
+
+def extract_converse_system_prompt(
+    messages: List[DialMessage],
+) -> ExtractSystemPromptResult:
+    system_msgs = []
+    found_non_system = False
+    system_messages_count = 0
+    modified_messages = []
+
+    for msg in messages:
+        if msg.role == DialRole.SYSTEM:
+            if found_non_system:
+                raise ValidationError(
+                    "A system message can only follow another system message"
+                )
+            system_messages_count += 1
+            match msg.content:
+                case str():
+                    system_msgs.append(msg.content)
+                case list():
+                    for part in msg.content:
+                        match part:
+                            case MessageContentTextPart():
+                                system_msgs.append(part.text)
+                            case MessageContentImagePart():
+                                raise ValidationError(
+                                    "System messages cannot contain images"
+                                )
+                case None:
+                    pass
+                case _:
+                    assert_never(msg.content)
+        else:
+            found_non_system = True
+            modified_messages.append(msg)
+    combined = "\n\n".join(msg for msg in system_msgs if msg)
+    return ExtractSystemPromptResult(
+        system_prompt=ConverseTextPart(text=combined) if combined else None,
+        system_message_count=system_messages_count,
+        modified_messages=modified_messages,
+    )
+
+
+async def process_messages(
```

Comment:
`process_messages` is very generic, it's like `do_something_with_messages` - it doesn't tell me much.

Maybe `to_converse_messages`? to point out how the messages change their representation.

Reply:
Renamed to `to_converse_messages`